base_layer/core/src/mempool/sync_protocol/initializer.rs

```diff
@@ -103,9 +104,12 @@ impl ServiceInitializer for MempoolSyncInitializer {
                 }
                 log_mdc::extend(mdc.clone());
             }
-
-            MempoolSyncProtocol::new(config, notif_rx, connectivity_event_subscription, mempool)
-                .run()
+            let streams = MempoolSyncStreams {
+                block_event_stream: base_node.get_block_event_stream(),
+                connectivity_events: connectivity_event_subscription,
+            };
+            MempoolSyncProtocol::new(config, notif_rx, mempool, connectivity)
+                .run(streams)
```

Comment:
nit: I'm not a fan of this pattern. Rather use the connectivity handle to get the event stream inside the run function.
```suggestion
           let base_node_events = base_node.get_block_event_stream();

            MempoolSyncProtocol::new(config, notif_rx, mempool, connectivity, base_node_events)
                .run()
```